tests/integration/ha/test_ha.py

```diff
@@ -213,3 +213,135 @@ async def test_two_clusters_not_replicated(ops_test: OpsTest, request):
     logger.info("Cleaning up old cluster...")
     await ops_test.model.applications[zk_2].remove()
     await helpers.wait_idle(ops_test)
+
+
+async def test_full_cluster_crash(ops_test: OpsTest, request):
+    hosts = helpers.get_hosts(ops_test)
+    leader_name = helpers.get_leader_name(ops_test, hosts)
+    leader_host = helpers.get_unit_host(ops_test, leader_name)
+    password = helpers.get_super_password(ops_test)
+    parent = request.node.name
+    non_leader_hosts = ",".join([host for host in hosts.split(",") if host != leader_host])
+
+    logger.info("Extending pebble restart delay on all units...")
+    helpers.modify_pebble_restart_delay(ops_test, policy="extend")
+
+    # letting the cluster settle
+    await helpers.wait_idle(ops_test)
+
+    logger.info("Starting continuous_writes...")
+    cw.start_continuous_writes(
+        parent=parent, hosts=hosts, username=helpers.USERNAME, password=password
+    )
+    await asyncio.sleep(CLIENT_TIMEOUT * 3)
+
+    logger.info("Counting writes are running at all...")
+    assert cw.count_znodes(
+        parent=parent, hosts=hosts, username=helpers.USERNAME, password=password
+    )
+
+    # kill all units "simultaneously"
+    await asyncio.gather(
+        *[
+            helpers.send_control_signal(ops_test, unit.name, signal="SIGKILL")
+            for unit in ops_test.model.applications[helpers.APP_NAME].units
+        ]
+    )
+
+    # Check that all servers are down at the same time before restoring them
+    try:
+        assert helpers.all_db_processes_down(ops_test), "Not all units down at the same time."
```

Comment:
**question** Not sure I understand why this assert is on a try block...?

Reply:
So that we can call `finally`. If the test fails, we don't 'necessarily' want to halt execution, but we still need to restore the pebble restart delay back to normal for future tests.